RadixWallet/Clients/DeviceFactorSourceClient/DeviceFactorSourceClient+Live.swift

```diff
@@ -71,68 +73,90 @@ extension DeviceFactorSourceClient: DependencyKey {
-				!userDefaults.getFactorSourceIDOfBackedUpMnemonics().contains($0.id)
-			}.map(\.id)
+				let mnemonicMissingFactorSources = factorSources
+					.filter(not(\.present))
+					.map(\.id)
 
-			func mnemonicMissing(_ account: Account) -> Bool {
-				switch account.securityState {
-				case let .unsecured(value):
-					mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
-				}
-			}
+				let mnemomincPresentFactorSources = factorSources
+					.filter(\.present)
+					.map(\.id)
 
-			func mnemonicMissing(_ persona: Persona) -> Bool {
-				switch persona.securityState {
-				case let .unsecured(value):
-					mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
-				}
-			}
+				let unrecoverableFactorSources = mnemomincPresentFactorSources
+					.filter { !backedUpFactorSources.contains($0) }
 
-			func unrecoverable(_ account: Account) -> Bool {
-				switch account.securityState {
-				case let .unsecured(value):
-					unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+				let network = try profile.network(id: profile.networkID)
+				let accounts = network.getAccounts()
+				let hiddenAccounts = network.getHiddenAccounts()
+				let personas = network.getPersonas()
+				let hiddenPersonas = network.getHiddenPersonas()
+
+				func mnemonicMissing(_ account: Account) -> Bool {
+					switch account.securityState {
+					case let .unsecured(value):
+						mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
 				}
-			}
 
-			func unrecoverable(_ persona: Persona) -> Bool {
-				switch persona.securityState {
-				case let .unsecured(value):
-					unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+				func mnemonicMissing(_ persona: Persona) -> Bool {
+					switch persona.securityState {
+					case let .unsecured(value):
+						mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
 				}
-			}
 
-			let mnemonicMissing = ProblematicAddresses(
-				accounts: accounts.filter(mnemonicMissing(_:)).map(\.address),
-				hiddenAccounts: hiddenAccounts.filter(mnemonicMissing(_:)).map(\.address),
-				personas: personas.filter(mnemonicMissing(_:)).map(\.address),
-				hiddenPersonas: hiddenPersonas.filter(mnemonicMissing(_:)).map(\.address)
-			)
+				func unrecoverable(_ account: Account) -> Bool {
+					switch account.securityState {
+					case let .unsecured(value):
+						unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
+				}
 
-			let unrecoverable = ProblematicAddresses(
-				accounts: accounts.filter(unrecoverable(_:)).map(\.address),
-				hiddenAccounts: hiddenAccounts.filter(unrecoverable(_:)).map(\.address),
-				personas: personas.filter(unrecoverable(_:)).map(\.address),
-				hiddenPersonas: hiddenPersonas.filter(unrecoverable(_:)).map(\.address)
-			)
+				func unrecoverable(_ persona: Persona) -> Bool {
```

Comment:
the two functions:

```swift
				func unrecoverable(_ account: Account) -> Bool {
					switch account.securityState {
					case let .unsecured(value):
						unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
					}
				}
```

and  `unrecoverable(_ account: Account)` can be merged into one:

```
func isUnrecoverable(entity: some EntityProtocol) -> Bool {
    switch entity.securityState {
         case .unsecured(control): unrecoverableFactorSources.contains(control.transactionSigning.factorSourceId)
    }
}
```

Reply:
Or by use of `AccountOrPersona`...